### PR TITLE
Clear cookies when the refresh token expires

### PR DESCRIPTION
**Pull request by gary-archer**
If the SPA is left overnight, this prevents the following type of behavior:
- Click Get API Data, results in a 401 getting a refresh token and a return to the unauthenticated view
- The unauthenticated view calls /login/end and still gets cookies so thinks the user is logged in

The response is written in the exception middleware so I add the header there.
I also updated the refresh token expiry test with some empty cookie checks.

### CODE REVIEW

File: src/main/kotlin/io/curity/oauthagent/CookieBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package io.curity.oauthagent
```
Comment (mtrojanowski) [inline src/main/kotlin/io/curity/oauthagent/CookieBuilder.kt:1]: I understand that this class was supposed to separate cookie encryption from serialization, right? It seems that currently it's not used anywhere.

**Review by mtrojanowski — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/curity/oauthagent/CookieBuilder.kt (hunk 1 above)
Comment (gary-archer) [reply src/main/kotlin/io/curity/oauthagent/CookieBuilder.kt]: Yep - accidental checkin - have deleted and will merge
